### DISCUSSION TITLE
Library asset encapsulation

### DISCUSSION THREAD

**Discussion by rpspall** [nrwl/nx#14872] [Feature Requests]
I have read documentation on how an Nx application can include assets from a library - makes sense how the application asset property (project.json) is configured with a reference to any library assets....

Call me old fashioned, but if I build a library (e.g. a React UI component library) has its own assets folder and is used by an application, shouldn't the library copy its own assets into the application's asset directory?

I would be nice if React component libraries could have their own asset directory and any dependent application automatically copies its assets into its own asset directory?

**Comment by kazeens** [comment]
@rpspall Hi
Have you found a solution to your problem?
